dhis-2/dhis-services/dhis-service-core/src/test/java/org/hisp/dhis/dataelement/DataElementWithValueTypeOptionsTest.java

```diff
@@ -0,0 +1,104 @@
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import org.hisp.dhis.DhisSpringTest;
+import org.hisp.dhis.analytics.AggregationType;
+import org.hisp.dhis.category.CategoryCombo;
+import org.hisp.dhis.common.FileTypeValueOptions;
+import org.hisp.dhis.common.ValueType;
+import org.hisp.dhis.common.ValueTypeOptions;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+public class DataElementWithValueTypeOptionsTest extends DhisSpringTest
+{
+    @Autowired
+    private DataElementStore dataElementStore;
+
+    @Test
+    public void testDeleteAndGetDataElement()
+    {
+        DataElement dataElementA = createDataElementWithValueTypeOptions( 'A', null );
+
+        dataElementStore.save( dataElementA );
+
+        long idA = dataElementA.getId();
+
+        assertNotNull( dataElementStore.get( idA ) );
+
+        dataElementA = dataElementStore.get( idA );
+
+        ValueTypeOptions valueTypeOptions = dataElementA.getValueTypeOptions();
+
+        assertNotNull( valueTypeOptions );
+        assertEquals( FileTypeValueOptions.class, valueTypeOptions.getClass() );
+        assertEquals( 100L, ((FileTypeValueOptions) valueTypeOptions).getMaxFileSize() );
+
+        dataElementStore.delete( dataElementA );
+
+        assertNull( dataElementStore.get( idA ) );
+    }
+
+    private DataElement createDataElementWithValueTypeOptions( char uniqueCharacter, CategoryCombo categoryCombo )
+    {
+        DataElement dataElement = new DataElement();
+        dataElement.setAutoFields();
+
+        dataElement.setUid( BASE_DE_UID + uniqueCharacter );
+        dataElement.setName( "DataElement" + uniqueCharacter );
+        dataElement.setShortName( "DataElementShort" + uniqueCharacter );
+        dataElement.setCode( "DataElementCode" + uniqueCharacter );
+        dataElement.setDescription( "DataElementDescription" + uniqueCharacter );
+        dataElement.setValueType( ValueType.FILE_RESOURCE );
+        dataElement.setDomainType( DataElementDomain.AGGREGATE );
+        dataElement.setAggregationType( AggregationType.SUM );
+        dataElement.setZeroIsSignificant( false );
+
+        if ( categoryCombo != null )
+        {
+            dataElement.setCategoryCombo( categoryCombo );
+        }
+        else if ( categoryService != null )
+        {
+            dataElement.setCategoryCombo( categoryService.getDefaultCategoryCombo() );
+        }
+
+        FileTypeValueOptions cvt = new FileTypeValueOptions();
+        cvt.setMaxFileSize( 100L );
```

Comment:
Since the test checks for the `100L` id think it helps if it is passed as parameter to this method so the value occurs visible in the test setup.